packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.ts

```diff
@@ -97,18 +108,23 @@ export class ConnectionTrackerService extends ImmutableStore<ConnectionTrackerSt
     this._statePersistenceService.saveConnectionTrackerState(this.state);
   }
 
+  removeItemsBelongingToRootCluster(clusterUri: string): void {
+    this.setState(draft => {
+      draft.connections = draft.connections.filter(i => {
+        const { rootClusterUri } =
+          this._trackedConnectionOperationsFactory.create(i);
```

Comment:
It feels like in the future each connection could have the `clusterUri` field instead so that we don't have to do the whole dance with `this._trackedConnectionOperationsFactory.create`.

Reply:
We already have some `resourceUri` in each connection (that contains `clusterUri`), so I wanted to avoid duplicating that information, but I can agree, it makes working with the connections a bit more difficult.

Reply:
But I'm fine with leaving it like this for now, let's just make sure to create such function when another opportunity to use it comes up.